historical_system_profiles/db_interface.py

```diff
@@ -39,6 +42,11 @@ def get_hsps_by_inventory_id(inventory_id, account_number):
     )
 
     query_results = query.all()
+    try:
+        ensure_correct_system_count([inventory_id], query_results)
```

Comment:
This code works but is not exactly right. The `ensure_correct_system_count` method is intended for use with IDs that map 1:1 with returned items. For example, if you sent in six profile IDs, you'd expect to get six profiles back.

In this usage, you're sending in one inventory ID, and expecting one or more profiles back.  The `ensure_correct_system_count` unintentionally works in this case.

I would change to something like this (untested pseudocode):

```python
if query.exists():
    return query_results
raise HTTPError(HTTPStatus.NOT_FOUND)
```